server/routes/surveyRoutes.js

```diff
@@ -22,36 +22,43 @@ module.exports = app => {
     res.send('Thanks for voting!');
   });
 
-  app.post('/api/surveys/webhooks', (req, res) => {
-    const p = new Path('/api/surveys/:surveyId/:choice');
-
-    _.chain(req.body)
-      .map(({ email, url }) => {
-        const match = p.test(new URL(url).pathname);
-        if (match) {
-          return { email, surveyId: match.surveyId, choice: match.choice };
-        }
-      })
-      .compact()
-      .uniqBy('email', 'surveyId')
-      .each(({ surveyId, email, choice }) => {
-        Survey.updateOne(
-          {
-            _id: surveyId,
-            recipients: {
-              $elemMatch: { email: email, responded: false }
+  app.post('/api/surveys/webhooks', async (req, res) => {
```

Comment:
This is the biggest change. This should really be moved to a service since it has lots of logic and writes to the database, but I left it for now so you could see the differences more easily.

First, is this should use async/await. It writes to the database, which is an async operation. The way it was written, if the database writes failed, you would never know about it. It would simply fire off a bunch of database calls, but instantly return an empty response. Since it's an async operation but without async/await, it returned the response of `{}` instantly, and then completed all the database writes.

Here I make it async so that it does not return a response until all writes are complete. If there is an error during a write, I simply log it to the console right now. In real life you'd probably want to also return an error response.